doc/build-helpers/build-helpers.chapter.md

```diff
@@ -0,0 +1,106 @@
+A build helper like this is said to accept **fixed-point arguments**.
+
+As [`stdenv.mkDerivation`'s support of fixed-point arguments](#mkderivation-recursive-attributes) was first included in the Nixpkgs release 22.05, `mkDerivation`-style build helpers don't always come with such support yet.doc/nixos
+
+### Defining a build helper with `lib.extendMkDerivation` {#ssec-build-helper-extendMkDerivation}
+
+The Nixpkgs library function `lib.customisation.extendMkDerivation` can be used to define a build helper with fixed-point arguments support from an existing one. It can also bring fixed-point argument support to existing builders as a slight wrapper around the original definition.
+
+The following is the conventional definition of build helper `mkLocalDerivation` that doesn't support fixed-point arguments:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-plain}
+
+# Example definition of `mkLocalDerivation` without fixed-point arguments support
+
+Write the followings in `pkgs/path/to/local-derivation/default.nix`:
+
+```nix
+{ lib
+, stdenv
+}:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+}@args:
+
+stdenv.mkDerivation (removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+and call the definition in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix):
+
+```nix
+{
+  mkLocalDerivation = callPackage ../path/to/local-derivation { };
+}
+```
+
+:::
+
+A slight wrapper of `lib.extendMkDerivation` around the original definition brings the magical `finalAttrs` to the builder:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-extendMkDerivation}
+
+# Example definition of `mkLocalDerivation` extended from `stdenv.mkDerivation` with `lib.extendMkDerivation`
+
+Write in `pkgs/path/to/local-derivation/default.nix`:
+
+```nix
+{ lib
+, stdenv
+}:
+
+lib.extendMkDerivation stdenv.mkDerivation (finalAttrs:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+}@args:
+
+removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+and call it the same way in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix).
+
+:::
+
+Should there be a need to modify the result derivation, `lib.customisation.extendMkDerivationModified` additionally takes a derivation modification function as the first argument. See the definition of `buildPythonModule`/`buildPythonApplication` in [`pkgs/development/interpreters/python/mk-python-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix) for practical use cases.
```

Comment:
Same here, don't link into internal Nixpkgs files from the user docs. If you just need it for an example you could link to a the file on a specific commit though.

Reply:
I see. The dependent commit hasn't been merged, and I cannot get a stable commit hash. I'll just remove the link.